jni/src/knn_extension/faiss/utils/BitSet.cpp

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "BitSet.h"
+
+FixedBitSet::FixedBitSet(const int* intArray, const int length){
+    int maxValue = intArray[length - 1];
+    this->n = (maxValue >> 6) + 1; // div by 64
+    this->bitmap = new uint64_t[this->n]();
+    for(int i = 0 ; i < length ; i ++) {
+        int value = intArray[i];
+        int bitsetArrayIndex = value >> 6;
+        bitmap[bitsetArrayIndex] |= 1ULL << value; // Equivalent of 1ULL << (value % 64)
```

Comment:
What is purpose of this?

Reply:
It is setting 1 in at bit position of the "value".